### ISSUE TITLE
Synology Download Station failed to add task from url

### ISSUE THREAD

**Issue by clappingmonkey** [Sonarr/Sonarr#1720]
Hi. (:

Title explains the error more or less. 

When Sonarr tries to download a torrent and add it to DownloadStation just can't. It returns these two errors (copying the whole text in case not to miss something): 
1. Request Failed. POST /api/release: Failed to add task from url magnet:?xt=urn:btih:4ec48675d528b3c9ad8831568669a42402399a52&dn=Legion.S01E01.1080p.HDTV.H264-TURBO%5Brartv%5D&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce. Reason: No default destination
2. Failed to add task from url magnet:?xt=urn:btih:4ec48675d528b3c9ad8831568669a42402399a52&dn=Legion.S01E01.1080p.HDTV.H264-TURBO%5Brartv%5D&tr=http%3A%2F%2Ftracker.trackerfix.com%3A80%2Fannounce&tr=udp%3A%2F%2F9.rarbg.me%3A2710&tr=udp%3A%2F%2F9.rarbg.to%3A2710&tr=udp%3A%2F%2Fopen.demonii.com%3A1337%2Fannounce. Reason: No default destination

That happens on either automated and manual download.

Sonarr is running on my Synology NAS. Model DS215j at DSM 6.1-15047. Sonarr version is 2.0.0.4613 Permissions are right.

Here`s the log file. I put it on pastebin: http://pastebin.com/ShnDzHZ2

**Comment by Taloth** [comment]
`Reason: No default destination`

Your synology download station has no default destination configured, so it doesn't know where to (temporarily) download it to. So you'll have to configure something.

@margaale Can we check for that in Test?

**Comment by margaale** [comment]
I guess we could. Have we look into the api...

**Comment by margaale** [comment]
@clappingmonkey did you log in and take those screenshots with same user that you used to configure the client in sonarr?

**Comment by margaale** [comment]
@clappingmonkey I was able to reproduce the behaviour you saw there. You have to login into the diskstation with that user and configure download station settings for that particular user.
@Taloth I've added the test.

**Comment by clappingmonkey** [comment]
@margaale nope, I took the shots when was logged with the admin user. But just a moment ago I enter with that user which Sonarr have and everything is just the same. I know what you mean with your question above. The situation when newly created user launch DownloadStation for first time and the app asks at startup to enter download location but that's not the case with me. I did that times ago. Last eve I had the idea to try how it works with the admin user. So I tried that. Firstly nothing seems to changed but during the night obviously that worked because this morning I saw that it downloaded few episodes. Today I'll reinstall Sonarr and DownloadStation and try it with a different newly created user.

**Comment by Taloth** [comment]
The check is merged, but not built yet. I prefer keeping the issue open till we find out if something else is wrong. Coz based on the responses, it might mean it takes time for the downloaddir setting to be applied in DS.

**Comment by margaale** [comment]
Sure, no problem, but when i tried yesterday the setting was applied immediately after clicking the ok button in my diskstation.

**Comment by clappingmonkey** [comment]
After reinstalling DownloadStation and Sonarr and tried with brand new user for the DS everything works now.

**Comment by margaale** [comment]
@clappingmonkey  Good to know. Did you update to latest development version before adding the client? (just asking to see if the additional check works?

**Comment by clappingmonkey** [comment]
No. But will do it asap for testing. 

Btw do you update your Synology to the latest update of DS? Version DSM 6.1-15047 Update 1? Because I had and after the reboot so the update can be applied Sonarr won't start. Maybe after the update the DS messed up something about Mono, because it`s not running also.

**Comment by margaale** [comment]
Nop, i haven't updated yet. Hope they haven't broken anything, Great hardware but the software is really crappy...
